stjude/proteinpaint
# Hic app refactor

## Description
The Hi-C app refactored into rx with several improvements

Devops changes:

1. App refactored to rx
2. Genome view in view model pattern
3. Reduced data request in genome view to one. Improved performance from 40s to roughly 6-8s on load
4. Controls includes stateless components to be reused in block code
5. New integration and unit tests

UI/UX changes:

1. New min cutoff
2. NONE shows as the default normalization method whether read from the file or not.
3. Able to launch chromosome pair view and detail via runproteinpaint()
4. Normalization method, cutoff off inputs, and matrix type hide in horizontal/two chromosome view
5. New loading component applies to appears for all views
6. Fixed bug in color scale
7. New wiki page and updated documentation link in Hi-C card. Note the wiki page only includes information about the track. When ready, information about the app will be added. 

To test:

- Pull `sjpp` hic.10 branch 
- Compare http://localhost:3000/?genome=hg19&hicfile=proteinpaint_demo/hg19/hic/hic_demo.hic against https://proteinpaint.stjude.org/?genome=hg19&hicfile=proteinpaint_demo/hg19/hic/hic_demo.hic. Please keep in mind, NONE shows by default. Change normalization method to VC to see a true comparison.
- Other Hi-C files available from http://localhost:3000/url.html
- Chromosome pair view from runproteinpaint(): http://localhost:3000/example.hic.chrpair.html
- Detail view from runproteinpaint(): http://localhost:3000/example.hic.detail.html
- Wiki page: https://github.com/stjude/proteinpaint/wiki/Hi%E2%80%90C 


## Checklist

[Check each task](https://github.com/stjude/proteinpaint/wiki/Pull-Request-Checklist) that has been performed or verified to be not applicable.
